src/models/linked_cell/CellGrid.cpp

```diff
@@ -0,0 +1,154 @@
+
+#include "CellGrid.h"
+#include "models/linked_cell/BoundaryCell.h"
+#include "models/linked_cell/Cell.h"
+#include "models/linked_cell/HaloCell.h"
+#include "models/linked_cell/InnerCell.h"
+#include <cmath>
+
+CellGrid::CellGrid(const std::array<double, 3>& domainSize, double cutoffRadius)
+    : domainSize(domainSize)
+    , cutoffRadius(cutoffRadius)
+    , gridDimensions({ 0, 0, 0 })
+{
+    initializeGrid();
+}
+
+CellGrid::~CellGrid() = default;
+
+void CellGrid::initializeGrid()
+{
+    for (int i = 0; i < 3; ++i) {
+        // Extra cells for halo
+        gridDimensions[i] = static_cast<size_t>(std::ceil(domainSize[i] / cutoffRadius)) + 2;
+    }
```

Comment:
Wenn wir das so machen wollen, benötigen wir unterschiedlich große Cells und ich glaube das ist schon etwas eklig. Ich würde lieber den cutoffRadius so vergrößern, so dass wir gleichgroße Zellen haben, die die Domain dann auch ausfüllen. Was sagst du?

Reply:
Hmm hatte mit gedacht dass leicht unterschiedliche Größen keinen wirklichen Unterschied machen, aber du hast recht es wäre schöner, wenn das alles einheitlich ist.

Reply:
Kann man nicht einfach abrunden und dann die Größen ausrechnen? Heißt alle sind gleich groß, aber ein wenig größer als cutoff
Dann hat man noch drei neue Werte, die jeweils die Größe einer Zelle in alle drei Dimensionen beschreibt.

Bsp.:
Domain Size = 5
Cutoff = 0.8
5 / 0.8 = 6,25 -> Wir brauchen 6 Cells

Die 6 Cells haben dann die Größe 5 / 6 = 0.833
Und das halt dann jeweils für alle drei Dimensionen. Die Größen kann man dann auch später nutzen.